include/swift/AST/DiagnosticsSema.def

```diff
@@ -3594,9 +3594,11 @@ ERROR(invalid_nonobjc_extension,none,
 ERROR(objc_in_extension_context,none,
       "members of constrained extensions cannot be declared @objc", ())
 ERROR(objc_in_generic_extension,none,
-      "members of extensions of "
-      "%select{classes from generic context|generic classes}0 "
-      "cannot be declared @objc", (bool))
```

Comment:
I think the distinction between "generic classes" and "classes defined in generic contexts" was serving a useful purpose, please put it back.

Reply:
I misread the code and thought it was setting it to true always (in 4.2 we had the restriction that classes with generic ancestry could not have categories attached). However it was different for a nested class inside a generic type, since isGeneric() is false in that case.

Reply:
Yeah, that's the intent—just showing we understand the code when talking to the user.